### ISSUE TITLE
Bug? `return true` and `return false` in function raise SC2152 (Can only return 0-255)

### ISSUE THREAD

**Issue by hoylemd** [koalaman/shellcheck#1701]
#### For bugs
- Rule Id: SC2152
- My shellcheck version: 0.7.0 
- [x] The rule's [wiki page](https://shellcheck.net/wiki/SC2152) does not already cover this
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```bash
#! /usr/bin/bash

my_fun() {
        return true;
}

if my_fun; then
        echo "hi";
fi
```

#### Here's what shellcheck currently says:

Can only return 0-255. Other data should be written to stdout. [SC2152] [sh/shellcheck]

#### Here's what I wanted or expected to see:

nothing

#### Additional comments

I trued to find an issue about this already, since it seems so basic to me, but I couldn't find anything.  The crux of the issue is that I prefer to use keywords and constants over hard-coded 'magic numbers', even if those numbers are just 0 and 1, so when I write a true-false check function, I like to make it do `return true` and `return false` for those cases.

When I do this, shellcheck complains that I'm not returning a number between 0 and 255, but `true` and `false` are aliases for `1` and `0` respectively (or so I understand - please correct me if I'm wrong here).

I know I can just disable that rule, but I do still want Shellcheck to ensure I'm returning status codes instead of random data.  I also know that I could just define `TRUE` and `FALSE` constants and use those (I checked, Shellcheck is fine with that), but typing `true` is a lot nicer than typing `$TRUE` every time.

**Comment by hoylemd** [comment]
Well now I feel like a dummy. `true` and `false` are *not* aliases for 0 and 1. It seems I can use them as keywords within a bash script. e.g.:

```bash
verbose=true;

if verbose; then
  echo 'true'';
fi

verbose=false;

if verbose; then
  echo 'false';
fi
```

will print out `true` but not `false`, but returning it from a function does seem to 'count' as a number. 

Closing this now, but if anyone has anything interesting or helpful to add, I'd appreciate the education!

**Comment by matthewpersico** [comment]
‘true’ and ‘false’ are commands whose only purpose is to exit 0 and exit 1, respectively. But you have to call them. Your syntax doesn’t call them. Try this (sorry, no backtick for formatting on a iPad keyboard):

verbose=foobar
if $verbose; then
    echo ‘bogus test’
fi 

😁

**Comment by matthewpersico** [comment]
Try this instead

true;  v=$?
if $v; then
    echo yep
else
    echo nope
fi
false;  v=$?
if $v; then
    echo yep
else
    echo nope
fi

**Comment by matthewpersico** [comment]
Ping @hoylemd, read the comments prior to this.

**Comment by matthewpersico** [comment]
You can use:

my_fun() {
        return $(true);
}

To get what you want.

**Comment by hoylemd** [comment]
@matthewpersico Thanks for the extra info! That's good stuff to know.
